### PR TITLE
Add API logging capabilities

### PR DESCRIPTION
**Pull request by ajacksified**
Add request log if your `debuglevel` is `info`.

:eyeglasses: @curioussavage 

(Sorry for the mess, also cleaned up a little whitespace.)

### CODE REVIEW

File: src/endpoints/v1.es6.js [hunk 1]
```diff
@@ -144,9 +155,20 @@ class APIv1Endpoint {
     });
   }
 
+  log (state, method, uri, options, status, err, duration) {
+    if (this.config.debugLevel === 'info') {
```
Comment (umbrae) [inline src/endpoints/v1.es6.js]: I guess we don't have integer equivalents for `>=` comparisons of debug level? Do we have a "debug" level for `debugLevel`? If so, shouldn't this also still run?

Comment (ajacksified) [inline src/endpoints/v1.es6.js]: Yeah... I suppose I should use an enum instead.

**Comment by curioussavage** [comment]
is there a pr for restcache too?

**Comment by ajacksified** [comment]
Correct. It may also make sense to log cache hits/misses as well, so maybe I'll follow up with one in restcache eventually too.

File: src/endpoints/v1.es6.js (hunk 1 above)
Comment (ajacksified) [inline src/endpoints/v1.es6.js]: Right now, it's all string based - I'll make a separate PR whereupon I overhaul the logging system.

**Comment by curioussavage** [comment]
:fish:

**Pull request merged**